packages/ember-headless-form/src/components/headless-form.gts

```diff
@@ -462,6 +462,18 @@ export default class HeadlessFormComponent<
     }
   }
 
+  @action
+  async onReset(e: Event): Promise<void> {
+    e.preventDefault();
+
+    for (const key of Object.keys(this.internalData)) {
+      delete this.internalData[key as keyof DATA];
+    }
+
+    this.validationState = undefined;
```

Comment:
Late to this, but: are we going to have explicit states for `pristine/dirty` or the consumer can figure this out from the `validationState` and the `submissionState`?